src/cmd/root.go

```diff
@@ -0,0 +1,224 @@
+
+			return nil
+		},
+		Version: version.GetVersion(),
+	}
+)
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		os.Exit(1)
+	}
+}
+
+func init() {
+	cobra.OnInitialize(initConfig)
+
+	rootCmd.PersistentFlags().BoolVarP(&rootFlags.assumeyes, "assumeyes", "y", false, "Automatically answer yes for all questions")
+	rootCmd.PersistentFlags().StringVar(&rootFlags.loglevel, "log-level", "warn", "Log messages above specified level: trace, debug, info, warn, error, fatal or panic")
+	rootCmd.PersistentFlags().BoolVar(&rootFlags.logPodman, "log-podman", false, "Show the log output of Podman. The log level is handled by the log-level option")
+	viper.BindPFlag("log-podman", rootCmd.PersistentFlags().Lookup("log-podman"))
+	// This flag is kept for compatibility reasons. In the future it would be better removed.
+	rootCmd.PersistentFlags().BoolVar(&rootFlags.verbose, "verbose", false, "Set log-level to 'debug'")
+	rootCmd.PersistentFlags().MarkDeprecated("verbose", "use 'log-level' instead.")
+}
+
+// initConfig reads in config file and ENV variables if set.
+func initConfig() {
+	if cfgFile != "" {
+		// Use config file from the flag.
+		viper.SetConfigFile(cfgFile)
+	} else {
+		// Find home directory.
+		home, err := os.UserHomeDir()
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+
+		// Search config in home directory with name ".toolbox" (without extension).
+		viper.AddConfigPath(home)
+		viper.SetConfigName(".toolbox")
+	}
+
+	viper.AutomaticEnv() // read in environment variables that match
+
+	// If a config file is found, read it in.
+	if err := viper.ReadInConfig(); err == nil {
+		fmt.Println("Using config file:", viper.ConfigFileUsed())
+	}
+}
+
+func setUpLoggers() error {
+	logrus.SetOutput(os.Stderr)
+	logrus.SetFormatter(&logrus.TextFormatter{
+		DisableTimestamp:       true,
+		DisableLevelTruncation: true,
+	})
+
+	if rootFlags.verbose {
+		rootFlags.loglevel = "debug"
+	}
+
+	lvl, err := logrus.ParseLevel(rootFlags.loglevel)
+	if err != nil {
+		return err
+	}
+
+	logrus.SetLevel(lvl)
+
+	return nil
+}
+
+func migrate() error {
+	configDir, err := os.UserConfigDir()
+	if err != nil {
+		return fmt.Errorf("Could not get the user config directory: %w", err)
+	}
+	toolboxConfigDir := fmt.Sprintf("%s/toolbox", configDir)
+	migrateStampPath := fmt.Sprintf("%s/podman-system-migrate", configDir)
+	logrus.Debugf("Toolbox config directory is %s", toolboxConfigDir)
+
+	podmanVersion, err := podman.GetVersion()
+	if err != nil {
+		return fmt.Errorf("Could not get the version of Podman: %w", err)
+	}
+	logrus.Debugf("Current Podman version is %s", podmanVersion)
+
+	err = os.MkdirAll(toolboxConfigDir, 0664)
+	if err != nil {
+		return fmt.Errorf("Configuration directory not created: %w", err)
+	}
+
+	output, err := ioutil.ReadFile(migrateStampPath)
```

Comment:
I wonder if we should run the contents of the file through [strings.TrimSpace](https://golang.org/pkg/strings/#TrimSpace)